### PR TITLE
Feature/ufs 242 int application setup

### PR DESCRIPTION
**Pull request by torchiad**

### CODE REVIEW

**Comment by torchiad** [comment]
Just adding an error message on validation fail

File: src/components/WorkflowComponentAdd/WorkflowComponentAdd.tsx [hunk 1]
```diff
@@ -1,12 +1,15 @@
-import React, { FC, useCallback } from "react";
```
Comment (torchiad) [inline src/components/WorkflowComponentAdd/WorkflowComponentAdd.tsx:1]: So consider this is the entry point @Darren-Ivey has used for his branch and the drag and drop. I would ideally review this along with the modification that have been made on his branch that will improve on this.

File: src/components/SetupApplication/SetupApplication.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            : "";
+
+    const opportunityId =
+        application &&
+        application.getApplication &&
+        application.getApplication.opportunity
+            ? application.getApplication.opportunity.id
+            : "";
+
+    const linkBack = `/setup/${opportunityId}`;
```
Comment (ascorbic) [inline src/components/SetupApplication/SetupApplication.tsx]: Is there a reason why you've pulled these out into separate variables?

File: src/components/SetupApplication/SetupApplicationForm.tsx [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    function validateHour(value: string) {
+        const intValue = parseInt(value);
+        if (!value || isNaN(intValue) || intValue > 23) {
+            return "Incorrect";
+        }
+    }
+
+    function validateMinute(value: string) {
+        const intValue = parseInt(value);
+        if (!value || isNaN(intValue) || intValue > 59) {
```
Comment (ascorbic) [inline src/components/SetupApplication/SetupApplicationForm.tsx:155]: Won't this fail on minute zero?

File: src/components/SetupOpportunity/SetupOpportunity.tsx [hunk 4]
```diff
@@ -38,7 +28,12 @@ export const SetupOpportunity: FC<Props> = ({ opportunity }) => {
         opportunity.getOpportunity.websiteListings &&
         opportunity.getOpportunity.websiteListings.items;
 
+    const hasApplications =
+        opportunity.getOpportunity.application &&
+        opportunity.getOpportunity.application.items;
+
     const websiteListings = opportunity.getOpportunity.websiteListings!.items;
+    const applications = opportunity.getOpportunity.application!.items;
```
Comment (ascorbic) [inline src/components/SetupOpportunity/SetupOpportunity.tsx]: These can be simplified in the same way you did above, by combining the test into the assigning a default, so something like:

```tsx
const websiteListings =         opportunity.getOpportunity.websiteListings &&
opportunity.getOpportunity.websiteListings.items.

File: src/components/SetupOpportunity/SetupOpportunity.tsx [hunk 5]
```diff
@@ -76,6 +71,7 @@ export const SetupOpportunity: FC<Props> = ({ opportunity }) => {
             </Details>
             <WorkflowComponentList
                 websiteListings={hasWebsiteListings ? websiteListings : null}
+                applications={hasApplications ? applications : null}
```
Comment (ascorbic) [inline src/components/SetupOpportunity/SetupOpportunity.tsx]: Just pass websiteListings, which will default to undefined, and change the typings for WorkflowCOmponentList.

File: src/components/WorkflowComponentList/WorkflowComponentList.tsx [hunk 6]
```diff
@@ -8,31 +8,47 @@ import { useMutation } from "react-apollo-hooks";
...
+    __typename: string;
+    id: string;
+    rank: number;
+    openApplication: string | null;
+    closeApplication: string | null;
+}
+
 interface Props {
     websiteListings: (WebsiteListing | null)[] | null;
+    applications: (Application | null)[] | null;
```
Comment (ascorbic) [inline src/components/WorkflowComponentList/WorkflowComponentList.tsx]: Change these types to match something like `applications?: Array<Application| null>`. Making it optional means that `| undefined` is implied.

File: src/components/SetupApplication/SetupApplicationForm.tsx (hunk 3 above)
Comment (torchiad) [reply src/components/SetupApplication/SetupApplicationForm.tsx:155]: It was ok here, it was further down the line when populating the page with 00's for the start of the hour.

File: src/components/SetupOpportunity/SetupOpportunity.tsx (hunk 4 above)
Comment (torchiad) [reply src/components/SetupOpportunity/SetupOpportunity.tsx]: Fixed in latest push

File: src/components/WorkflowComponentList/WorkflowComponentList.tsx (hunk 6 above)
Comment (torchiad) [reply src/components/WorkflowComponentList/WorkflowComponentList.tsx]: Made these changes

File: src/components/SetupOpportunity/SetupOpportunity.tsx (hunk 5 above)
Comment (torchiad) [reply src/components/SetupOpportunity/SetupOpportunity.tsx]: Fixed

File: src/components/SetupApplication/SetupApplicationForm.tsx [hunk 7]
```diff
@@ -36,13 +36,27 @@ export const SetupApplicationForm: FC<Props> = ({
             openMonth: openDate.getMonth() + 1 || "",
             openDay: openDate.getDate() || "",
             openHour: openDate.getHours() || "",
-            openMinute: openDate.getMinutes() || "",
+            openMinute:
```
Comment (ascorbic) [inline src/components/SetupApplication/SetupApplicationForm.tsx]: Can you refactor this. These sort of nested ternaries are really hard to read, and it seems a very convulated way to just zero-pad a number.

File: src/components/SetupApplication/SetupApplicationForm.tsx [hunk 8]
```diff
@@ -36,13 +36,27 @@ export const SetupApplicationForm: FC<Props> = ({
...
+                    ? `0${openDate.getMinutes()}`
+                    : "",
 
             closeYear: closeDate.getFullYear() || "",
             closeMonth: closeDate.getMonth() + 1 || "",
             closeDay: closeDate.getDate() || "",
             closeHour: closeDate.getHours() || "",
-            closeMinute: closeDate.getMinutes() || ""
+            closeMinute:
+                closeDate.getMinutes() === 0
```
Comment (ascorbic) [inline src/components/SetupApplication/SetupApplicationForm.tsx]: Same here. Don't repeat yourself.

File: src/components/SetupOpportunity/SetupOpportunity.tsx [hunk 9]
```diff
@@ -23,18 +23,18 @@ export const SetupOpportunity: FC<Props> = ({ opportunity }) => {
...
-        opportunity.getOpportunity.websiteListings.items;
+        opportunity.getOpportunity.websiteListings.items
+            ? opportunity.getOpportunity.websiteListings.items
+            : undefined;
 
-    const hasApplications =
+    const applications =
         opportunity.getOpportunity.application &&
-        opportunity.getOpportunity.application.items;
+        opportunity.getOpportunity.application.items
```
Comment (ascorbic) [inline src/components/SetupOpportunity/SetupOpportunity.tsx]: No need for the ternary here: just do `|| null`

**Review by ascorbic — CHANGES_REQUESTED** [review]

File: src/components/SetupApplication/SetupApplicationForm.tsx (hunk 7 above)
Comment (torchiad) [reply src/components/SetupApplication/SetupApplicationForm.tsx]: I've moved it into a function

File: src/components/SetupApplication/SetupApplicationForm.tsx (hunk 8 above)
Comment (torchiad) [reply src/components/SetupApplication/SetupApplicationForm.tsx]: As above  - "I've moved it into a function"

File: src/components/SetupOpportunity/SetupOpportunity.tsx (hunk 9 above)
Comment (torchiad) [reply src/components/SetupOpportunity/SetupOpportunity.tsx]: fixed

**Review by ascorbic — APPROVED** [review]

**Pull request merged**